### PR TITLE
Bug timed window

### PR DESCRIPTION
**Pull request by snth**
Very hacky workaround for the timed_window AttributeError.

It seems to me that the `AsyncIOMainLoop` in `tornado` doesn't implement `loop._running` properly (it's not implemented at all). Given that it starts with an underscore, I don't know if it forms part of the API and needs to be implemented?

This fixes the `fib_asyncio.py` example. However in the application where I discovered this and where I am trying to use it my whole app now freezes. Still investigating what's going on there. It would be good to get some more eyes on this to see if my workaround works for other people.

### CODE REVIEW

**Comment by snth** [comment]
This should hopefully fix #131.

File: examples/fib_asyncio.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 L = s.sink_to_list()                    # store result in a list
 
 s.rate_limit(0.5).sink(source.emit)         # pipe output back to input
-s.rate_limit(1.0).sink(lambda x: print(L))  # print state of L every second
+s.timed_window(1.0).sink(lambda x: print(x))
```
Comment (mrocklin) [inline examples/fib_asyncio.py]: Did you mean to change this in the PR?

File: streamz/core.py [hunk 2]
```diff
@@ -1130,7 +1130,14 @@ def make_coro():
...
 
-    if not loop._running:
+    try:
+        loop_is_running = loop._running
+    except AttributeError:
+        try:
+            loop_is_running = loop.asyncio_loop.is_running()
+        except AttributeError:
+            raise NotImplementedError(type(loop))
+    if not loop_is_running:
```
Comment (mrocklin) [inline streamz/core.py]: I recommend removing this code rather than protecting it.

File: examples/fib_asyncio.py (hunk 1 above)
Comment (snth) [reply examples/fib_asyncio.py]: Yes. It changes the functionality a bit but now this example tests the rate_limit() and timed_window() methods. I've been using this to test the other changes I've been making in this PR.

File: streamz/core.py (hunk 2 above)
Comment (snth) [reply streamz/core.py]: Ok thanks for the pointer. I've done that. The fib_asyncio.py example runs fine now. I tried running the tests but I'm having trouble with my setup on Windows. Travis also seems to be reporting a failing check.

**Comment by snth** [comment]
Unfortunately I doubt I'll be able to add any tests. Sorry about the file permissions. That was not intended. My setup on Windows under MobaXterm is not ideal. I'm not sure why the permissions were changed.

I removed the `sync` method. I hope that is what you intended. If you could please test it on your side and maybe let me know what's causing the Travis checks to fail.

**Comment by mrocklin** [comment]
You can see what tests are failing by clicking on the details link for the travis-ci failure.  Here is a direct link: https://travis-ci.org/mrocklin/streamz/jobs/312773565

**Comment by martindurant** [comment]
`loop._running` was removed here https://github.com/mrocklin/streamz/pull/152/files#diff-d4e512993f47710ea9e7155d66011a24L1177 , along with a firmer view on event loops and threads.

`fib_asyncio.py` appears to work OK on this end, so perhaps this PR has actually be resolved?

**Comment by snth** [comment]
Thanks for the ping. I'll try to rebase on a recent master in the next few days and check if it is resolved.

**Comment by martindurant** [comment]
@snth , is this still an issue? Would you like to base on current master, where things are generally passing again?

**Pull request closed without merging**